app/models/pawn.rb

```diff
@@ -49,6 +49,38 @@ def valid_move?(x,y)
     return false
   end
 
+  # returns true if unfriendly pawn on square
+  def unfriendly_pawn?(x,y)
+    piece = self.game.find_piece(x,y)
+    return (piece.type == "Pawn") && ((piece.white? != white?) || (piece.black? != black?))
```

Comment:
You actually only need `piece.white? != white?` (or `piece.black? != black?`) because of the following:


`self` | `piece` | `piece.black?` | `black?` | `piece.black? != black?`
--- | --- | --- | --- | ---
White | White | `false` | `false` | `true`
White | Black | `false` | `true` | `false`
Black | White | `true` | `false` | `false`
Black | Black | `true` | `true` | `true`

tl;dr - you only need one